### PR TITLE
etcdmain: add log-output flag

### PR DESCRIPTION
**Pull request by gyuho**
So we can choose where to write logs.
Fix https://github.com/coreos/etcd/issues/5449.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @xiang90 @heyitsanthony @alban 

Please review. Thanks.

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -616,6 +617,42 @@ func setupLogging(cfg *config) {
...
+				return
+			}
+			capnslog.SetFormatter(capnslog.NewPrettyFormatter(f, cfg.Debug))
+		}
+	}
+}
+
+func openToAppend(fpath string) (*os.File, error) {
+	f, err := os.OpenFile(fpath, os.O_RDWR|os.O_APPEND, fileutil.PrivateFileMode)
+	if err != nil {
```
Comment (heyitsanthony) [inline etcdmain/etcd.go]: not needed if using os.O_CREATE

File: etcdmain/etcd.go [hunk 2]
```diff
@@ -616,6 +617,42 @@ func setupLogging(cfg *config) {
...
+	case "journald":
+		f, err := capnslog.NewJournaldFormatter()
+		if err != nil {
+			plog.Warningf("couldn't log to journald (%v)", err)
+			return
+		}
+		capnslog.SetFormatter(f)
+
+	default:
+		if filepath.Ext(cfg.LogOutput) == ".log" {
```
Comment (heyitsanthony) [inline etcdmain/etcd.go]: don't bother checking for .log? not sure if silently failing into stderr is the right behavior here

File: etcdmain/etcd.go (hunk 1 above)
Comment (gyuho) [inline etcdmain/etcd.go]: Ok will fix. thanks

File: etcdmain/etcd.go [hunk 3]
```diff
@@ -616,6 +617,42 @@ func setupLogging(cfg *config) {
 		}
 		repoLog.SetLogLevel(settings)
 	}
+	switch cfg.LogOutput {
+	case "", "stderr":
```
Comment (heyitsanthony) [inline etcdmain/etcd.go]: does journald auto detection still work?

File: etcdmain/etcd.go (hunk 2 above)
Comment (gyuho) [inline etcdmain/etcd.go]: Ok will remove the extension checking.

Comment (xiang90) [inline etcdmain/etcd.go]: probably just remove the log file. user can easily redirect if they want.

**Comment by gyuho** [comment]
Closing to see if we can find easier ways.

**Pull request closed without merging**